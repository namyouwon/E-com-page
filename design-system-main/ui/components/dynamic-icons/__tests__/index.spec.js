/* eslint-env jest */
import React from 'react';

import TypingIcon from '../typing';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

describe('Typing icon', () => {
  it('renders a typing icon', () =>
    matchesMarkup(
      <TypingIcon assistiveText="User typing" title="User typing" />
    ));

  it('renders an animated typing icon', () =>
    matchesMarkup(
      <TypingIcon assistiveText="User typing" isAnimated title="User typing" />
    ));

  it('renders a paused animated typing icon', () =>
    matchesMarkup(
      <TypingIcon
        assistiveText="User typing"
        isAnimated
        isPaused
        title="User typing"
      />
    ));
});
