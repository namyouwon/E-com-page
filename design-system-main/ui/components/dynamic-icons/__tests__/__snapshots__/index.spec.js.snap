// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typing icon renders a paused animated typing icon 1`] = `
<span
  className="slds-icon-typing slds-is-animated slds-is-paused"
  title="User typing"
>
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-assistive-text"
  >
    User typing
  </span>
</span>
`;

exports[`Typing icon renders a typing icon 1`] = `
<span
  className="slds-icon-typing"
  title="User typing"
>
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-assistive-text"
  >
    User typing
  </span>
</span>
`;

exports[`Typing icon renders an animated typing icon 1`] = `
<span
  className="slds-icon-typing slds-is-animated"
  title="User typing"
>
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-icon-typing__dot"
  />
  <span
    className="slds-assistive-text"
  >
    User typing
  </span>
</span>
`;
